src/pg_tde_event_capture.c

```diff
@@ -113,7 +113,7 @@ pg_tde_ddl_command_start_capture(PG_FUNCTION_ARGS)
 
 		tablespace_oid = stmt->tablespacename != NULL ? get_tablespace_oid(stmt->tablespacename, false) 
 						 : MyDatabaseTableSpace;  
-		principal_key = GetPrincipalKey(MyDatabaseId, tablespace_oid);
+		principal_key = GetPrincipalKey(MyDatabaseId, tablespace_oid, LW_SHARED);
```

Comment:
```suggestion
		LWLockAcquire(tde_lwlock_enc_keys(), LW_SHARED);
		principal_key = GetPrincipalKey(MyDatabaseId, tablespace_oid, LW_SHARED);
		LWLockRelease(tde_lwlock_enc_keys());
```
_The caller must hold a `tde_lwlock_enc_keys` lock and pass its obtained mode via the `lockMode` param_